addons/irc/irc.lua

```
@@ -0,0 +1,195 @@
+--[[
+
+Please edit the ./data/settings.xml to set username and password.
+
+This is a very early version of a IRC based Linkshell chat addon.
+It currently works fine with most IRC servers even those that require authentification.
+It was aminly developed to talk with people on twitch.tv that watch your stream without
+the need to have a second monitor to view the chat.
+
+Suggestions are very welcome.
+Skyrant
+
+]]
+
+socket = require "socket"
+
+_addon.version = '0.1'
+_addon.name = 'irc'
+_addon.author = 'Skyrant, maintainer: Skyrant'
+_addon.commands = {'i','irc'}
+
+require('chat')
+require('logger')
+require('strings')
+require('maths')
+require('tables')
+
+config = require('config')
+settings = config.load(defaults)
+
+ircconnect = function()
+
+	connecttime = os.time()
+
+	serveraddr = settings.server.serveraddr
+	serverport = settings.server.serverport
+
+	if serveraddr and serverport then
+		-- eventually do something here
+	else
+		log("setting default twitch server and port")
+		serveraddr = 'irc.twitch.tv'
```

I really think it should abort instead of defaulting to _Twitch_. That will be entirely unexpected to users just wanting an IRC addon and suddenly they'll find themselves in a chatroom they had no intention of joining.